### ISSUE TITLE
Runbot unavailable in v11

### ISSUE THREAD

**Issue by jbeficent** [OCA/maintainer-quality-tools#503]
Hi,

Runbot is only available, for 11.0 branches, in OCB repo. Does anyone have a clue why?

We are pushing very hard to migrate modules to v11 and it is fustrating with the current situation.

We can put resources on the topic, but we need aome initial guidance.

Perhaps @moylop260 and @gurneyalex know?

**Comment by moylop260** [comment]
OCB is not using travis2docker.
In fact, it is using the default runbot behaviour so installation of py3.5
and requirements is required from the host

Alexandre Could you help us?
-- 
Moisés López Calderón
Vauxoo - OpenERP's Gold Partner
Mobile: (+521) 477-752-22-30
Office: (+52) 477-773-33-46
web: http://www.vauxoo.com
twitter: @vauxoo
           @moylop260
hangout: moylop260@vauxoo.com

**Comment by pedrobaeza** [comment]
If I activate travis2docker it would work?

**Comment by moylop260** [comment]
Yes is it. I remember that it was disabled because the new branches of OCB don't have `.travis.yml` file by default so runbot with t2d fails.

But we need first create the `.travis.yml` file for all branches and if a new one is created so adding the `.travis.yml` too.

For example: https://github.com/OCA/OCB/blob/10.0/.travis.yml

**Comment by lasley** [comment]
@jbeficent - to confirm, you're saying that only OCB v11 Runbot works, correct? Can you point to some good failing samples?

**Comment by nhomar** [comment]
> Yes is it. I remember that it was disabled because the new branches of OCB don't have .travis.yml file by default so runbot with t2d fails.

> But we need first create the .travis.yml file for all branches and if a new one is created so adding the .travis.yml too.

> For example: https://github.com/OCA/OCB/blob/10.0/.travis.yml

Why is that? should not this be added by default?

**Comment by nhomar** [comment]
ping @moylop260

**Comment by moylop260** [comment]
Yes
It should be added when a new branch is added from odoo/odoo to oca/ocb but maybe is missed by the responsible

**Comment by pedrobaeza** [comment]
An attempt of adding the file is here: https://github.com/OCA/OCB/pull/664

**Comment by lasley** [comment]
I still think we're diagnosing the wrong problem here. @jbeficent [can you please confirm whether you were reporting that the OCB branch is broken, or all other branches](https://github.com/OCA/maintainer-quality-tools/issues/503#issuecomment-340296854)? From what I can see, OCB was broken entirely by OCA/OCB#664, but I think the wrong issue was being diagnosed in the first place.

**Comment by moylop260** [comment]
You are right!
I'll create a new PR to enable it
Thanks!

**Comment by lasley** [comment]
My point still stands- I think Jordi was reporting that branches other than OCB are failing on v11:

> Runbot is only available, for 11.0 branches, in OCB repo.

My interpretation of the above is that Runbot was only available on OCB/11.0.

**Comment by jbeficent** [comment]
@lasley That is exactly the case. OCB branch is available with runbot, but all other branches not.

**Comment by pedrobaeza** [comment]
I reiterate the need of Travis + runbot dockerized with other method (Doodba or whatever), but avoiding all the problems we have right now (we have a lot OCA repositories stuck due to wkhtmltopdf issue that locks the tests).

**Comment by jbeficent** [comment]
See how all other branches are unavailable.
![image](https://user-images.githubusercontent.com/7683926/32607671-047527a6-c55a-11e7-9cd6-69645dbf97a5.png)

**Comment by jbeficent** [comment]
@moylop260 @pedrobaeza At this stage I have not clue on what is really happening. @pedrobaeza how do you know it is an issue with wkhtmltopdf?

**Comment by lasley** [comment]
Yeah we're hitting major growing pains right now - we have about 5 issues that are cropping up and causing me to pull out a lot of hair.

This isn't wkhtmltopdf. I'm looking now

**Comment by lasley** [comment]
The issue is specific [to v11](https://runbot.odoo-community.org/runbot/static/build/3309657-11-0-273557/logs/job_30_run.txt)

```

Testing l10n_es_partner,l10n_es_account_bank_statement_import_n43,l10n_es_partner_mercantil,l10n_es_account_asset,l10n_es_account_invoice_sequence:
createdb: database creation failed: ERROR:  database "openerp_test" already exists
/home/odoo/OCB-11.0/odoo-bin -d openerp_test --db-filter=^openerp_test$ --log-level info --pidfile=/tmp/odoo.pid
Traceback (most recent call last):
File "/home/odoo/OCB-11.0/odoo-bin", line 5, in <module>
import odoo
File "/.repo_requirements/odoo/odoo/__init__.py", line 84, in <module>
from . import modules
File "/.repo_requirements/odoo/odoo/modules/__init__.py", line 8, in <module>
from . import db, graph, loading, migration, module, registry
File "/.repo_requirements/odoo/odoo/modules/graph.py", line 10, in <module>
import odoo.tools as tools
File "/.repo_requirements/odoo/odoo/tools/__init__.py", line 8, in <module>
from .config import config
File "/.repo_requirements/odoo/odoo/tools/config.py", line 648, in <module>
config = configmanager()
File "/.repo_requirements/odoo/odoo/tools/config.py", line 316, in __init__
self._parse_config()
File "/.repo_requirements/odoo/odoo/tools/config.py", line 386, in _parse_config
self.load()
File "/.repo_requirements/odoo/odoo/tools/config.py", line 526, in load
p.read([self.rcfile])
File "/usr/lib/python3.5/configparser.py", line 696, in read
self._read(fp, filename)
File "/usr/lib/python3.5/configparser.py", line 1089, in _read
fpname, lineno)
configparser.DuplicateOptionError: While reading from '/home/odoo/.openerp_serverrc' [line  5]: option 'addons_path' in section 'options' already exists
```

**Comment by pedrobaeza** [comment]
@jbeficent I'm not talking that whtmltopdf (or whatever) issue is related with v11, but that we need a powerful system and abstraction layer for avoiding these problems or at least make them more easily fixable. And as extra, we will have solved in a very simple way the current problem of v11.

**Comment by lasley** [comment]
Looks like the config is duplicated somewhere. It seems that Odoo v11 now checks for dup keys, while Odoo v-previous simply overwrote the keys.
```
(nodejs)(python3.5) odoo@2a7d03b5a7fe:~/build/LasLabs/web$ cat /home/odoo/.openerp_serverrc
[options]
addons_path = /home/odoo/build/LasLabs/web,/home/odoo/OCB-11.0/addons
data_dir = /home/odoo/data_dir

addons_path = /home/odoo/build/LasLabs/web,/home/odoo/OCB-11.0/addons
data_dir = /home/odoo/data_dir
```

Tracked to [here](https://github.com/OCA/maintainer-quality-tools/blob/master/travis/test_server.py#L276). Thinking of fix

**Comment by moylop260** [comment]
I have created the issue related with: https://github.com/OCA/maintainer-quality-tools/issues/508
Thanks @lasley

**Comment by moylop260** [comment]
> @pedrobaeza  
we have a lot OCA repositories stuck due to wkhtmltopdf issue that locks the tests

But feel free of create a issue from https://github.com/OCA/maintainer-quality-tools/issues/new because I don't know it

**Comment by pedrobaeza** [comment]
Independently from that, each issue is very time consuming for debugging/repairing. We need a better isolated base.

**Comment by moylop260** [comment]
You don't need debugging/repairing a issue, you just need create it and the MQT expert will debug and repair it for you.
But if you don't create a issue we won't fix it because we won't know it.

> We need a better isolated base.
+1 totally agree
MQT is ugly and dirty we need clean it and make it a good structure and create it for all platforms (not just travis).
We will work on that too.

**Comment by jbeficent** [comment]
@moylop260 So with #508 you are clear on what you have to fix?

**Comment by pedrobaeza** [comment]
Here it is the other issue, Moi: https://github.com/OCA/maintainer-quality-tools/issues/509

**Comment by moylop260** [comment]
@jbeficent 
Yes, It's because is appending instead of updating... The fix is https://github.com/OCA/maintainer-quality-tools/pull/510

**Comment by moylop260** [comment]
The fix was merged and the runbot 11.0 is now available (It isn't a wkhtmltopdf issue related, but the other issue unrelated here was fixed too https://github.com/OCA/contract/pull/116)

<img width="508" alt="screen shot 2017-11-09 at 23 23 19" src="https://user-images.githubusercontent.com/6644187/32644080-053161be-c5a5-11e7-83bd-50deac85fab1.png">

**Comment by jbeficent** [comment]
Thank you very much!!! @moylop260
